lib/providers/memory_history.dart

```diff
@@ -46,6 +46,10 @@ class MemoryHistory implements HistoryProvider {
     if (!linkMatcher.matches(anchor)) {
       return;
     }
+    if (anchor.target != null && anchor.target.isNotEmpty) {
+      // Prevent router from swallowing links that should open in another window
+      return;
+    }
```

Comment:
I don't really like how this is repeated three times, but I am following the pattern that is already present here for clarity.